### PR TITLE
fix: encode email with encodeUriComponent before searching

### PR DESCRIPTION
**Pull request by ItsSudip**
## Description of the change

Adding URL encoding of email before user lookup call. 

## Type of change
- [X] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by utsabc** [comment]
Lets also get the E2E test report as well

**Review by utsabc — APPROVED** [review]

File: test/__tests__/data/salesforce_router_output.json [hunk 1]
```diff
@@ -57,6 +57,64 @@
...
+        "method": "POST",
+        "endpoint": "https://ap15.salesforce.com/services/data/v50.0/sobjects/Lead/leadId?_HttpMethod=PATCH",
+        "headers": {
+          "Content-Type": "application/json",
+          "Authorization": "Bearer 00D2v000002lXbX!ARcAQJBSGNA1Rq.MbUdtmlREscrN_nO3ckBz6kc4jRQGxqAzNkhT1XZIF0yPqyCQSnezWO3osMw1ewpjToO7q41E9.LvedWY"
+        },
+        "params": {},
+        "body": {
+          "JSON": {
+            "Email": "ddv_ua+{{1234*245}}@bugFix.com",
```
[the remark this answers is not in the document]
Comment (ItsSudip) [reply test/__tests__/data/salesforce_router_output.json:74]: Yes, we are encoding for lookup calls only not for sending the data we don't need to URL encode as we are sending inside the body.

File: src/v0/destinations/salesforce/transform.js [hunk 2]
```diff
@@ -217,7 +217,9 @@ async function getSalesforceIdFromPayload(
     // its a lead object. try to get lead object id using search query
     // check if the lead exists
     // need to perform a parameterized search for this using email
-    const email = getFieldValueFromMessage(message, "email");
+    const email = encodeURIComponent(
+      getFieldValueFromMessage(message, "email")
```
Comment (koladilip) [inline src/v0/destinations/salesforce/transform.js:221]: did we test with saleforce using real data?

Comment (ItsSudip) [reply src/v0/destinations/salesforce/transform.js:221]: Yes

**Pull request merged**